frontend/views/pages/PendingApproval.vue

```diff
@@ -54,15 +55,35 @@ export default ({
   },
   mounted () {
     this.ephemeral.groupIdWhenMounted = this.currentGroupId
-    sbp('chelonia/contract/wait', this.ourIdentityContractId).then(async () => {
-      await sbp('chelonia/contract/sync', this.ephemeral.groupIdWhenMounted)
+    let reset = false
+    let destroyed = false
+
+    const syncPromise = sbp('chelonia/contract/wait', this.ourIdentityContractId).then(async () => {
+      if (destroyed) return
+      reset = false
+      await sbp('chelonia/contract/retain', this.ephemeral.groupIdWhenMounted, { ephemeral: true })
       this.ephemeral.contractFinishedSyncing = true
       if (this.haveActiveGroupProfile) {
         this.ephemeral.groupJoined = true
       }
     }).catch(e => {
       console.error('[PendingApproval.vue]: Error waiting for contract to finish syncing', e)
     })
+    const listener = () => { reset = true }
+    this.ephemeral.ondestroy = () => {
+      destroyed = true
+      sbp('okTurtle.events/off', CHELONIA_RESET, listener)
+      syncPromise.finally(() => {
+        if (reset) return
+        sbp('chelonia/contract/release', this.ephemeral.groupIdWhenMounted, { ephemeral: true }).catch(e => {
+          console.error('[PendingApproval.vue]: Error releasing contract', e)
+        })
+      })
+    }
+    sbp('okTurtle.events/on', CHELONIA_RESET, listener)
+  },
+  beforeDestroy () {
+    this.ephemeral.ondestroy?.()
```

Comment:
This isn't really new logic, rather redefining the old logic that called `/sync`, which is unsafe to call.

> It's possible that beforeDestroy could be called as a result of the group switcher switching to a different group

I don't see how that'd break anything. The `onDestroy` is there to call `release`, which is the correct thing to do after calling `retain`. If the view is destroyed, there's no point in keeping the group around (if it shouldn't be released, the reference count should indicate that). Note that before this only called `sync` and it also didn't affect the reference count.

> did you test this with multiple groups?

No, I'm not sure how that'd be done as joining a group would open a new URL?

Reply:
> No, I'm not sure how that'd be done as joining a group would open a new URL?

You'd create and join group1 as u1. Then u2 (in a separate container tab or browser) would send you an invite link to group2, but before u1 visits it, u2 would log out. Then u1 visits the link to join u2's group2 and get stuck on this pending page. Then they'd switch back to their own group. Then u2 would log back in.

> I don't see how that'd break anything. The onDestroy is there to call release, which is the correct thing to do after calling retain. If the view is destroyed, there's no point in keeping the group around (if it shouldn't be released, the reference count should indicate that). Note that before this only called sync and it also didn't affect the reference count.

I see. Could you add a comment here as well then to explain where the "real" (persistent) retain happens on this group and why we're calling ephemeral retain here instead of `sync`?

Reply:
> You'd create and join group1 as u1. [...]

I did this and nothing broke, as expected, since this is merely a UI concern.

> Could you add a comment here as well then

Done. Also added a comment to the effect that `vue` files should not be managing references.